source/xprod.js

```diff
@@ -24,11 +24,11 @@ var xprod = _curry2(function xprod(a, b) { // = xprodWith(prepend); (takes about
   var ilen = a.length;
   var j;
   var jlen = b.length;
-  var result = [];
+  var result = Array(ilen + jlen);
   while (idx < ilen) {
     j = 0;
     while (j < jlen) {
-      result[result.length] = [a[idx], b[j]];
+      result[idx + j] = [a[idx], b[j]];
```

Comment:
And here in the inner loop, we are adding `jlen` elements each time through, so I believe we have

`result[(idx * jlen) + j] = [a[idx], b[j]];`

It is also a bit confusing to read with `idx` and `j` as the index variables. Can we update this function to use just plain old `i` to match the `j` (I know you didn't name these originally)

Reply:
Updated